edx_notifications/stores/sql/store_provider.py

```diff
@@ -395,3 +395,111 @@ def get_all_active_timers(self, until_time=None, include_executed=False):
             objs = objs.filter(executed_at__isnull=True)
 
         return [obj.to_data_object() for obj in objs]
+
+    def get_notification_preference(self, name):
+        """
+        Will return a single NotificationPreference if exists
+        else raises exception ItemNotFoundError
+        """
+        try:
+            obj = SQLNotificationPreference.objects.get(name=name)
+        except ObjectDoesNotExist:
+            raise ItemNotFoundError()
+
+        return obj.to_data_object()
+
+    def save_notification_preference(self, notification_preference):
+        """
+        Will save (create or update) a NotificationPreference in the
+        StorageProvider
+        """
+        obj = None
+        if notification_preference.name:
+            # see if it exists
+            try:
+                obj = SQLNotificationPreference.objects.get(name=notification_preference.name)
+                obj.load_from_data_object(notification_preference)
+            except ObjectDoesNotExist:
+                pass
+        if not obj:
+            obj = SQLNotificationPreference.from_data_object(notification_preference)
+
+        obj.save()
+        return obj.to_data_object()
+
+    def get_all_notification_preferences(self):
+        """
+        This returns list of all registered NotificationPreference.
+        """
+        query = SQLNotificationPreference.objects.all()
+
+        result_set = [item.to_data_object() for item in query]
+
+        return result_set
+
+    def get_user_preference(self, user_id, name):
+        """
+        Will return a single UserNotificationPreference if exists
+        else raises exception ItemNotFoundError
+        """
+        try:
+            obj = SQLUserNotificationPreferences.objects.get(user_id=user_id, preference__name=name)
+        except ObjectDoesNotExist:
+            raise ItemNotFoundError()
+
+        return obj.to_data_object()
+
+    def set_user_reference(self, user_preference):
```

Comment:
typo: this should be set_user_preference(). This will also have to be updated in tests as well.